### PR TITLE
Support Array Of Fields For Custom Transformation

### PR DESCRIPTION
**Pull request by dduan**
Returns `nil` if key doesn't exist or transformation for found value throws.

### CODE REVIEW

**Comment by dduan** [comment]
@keith

File: Sources/Mapper.swift [hunk 1]
```diff
@@ -346,6 +346,23 @@ public struct Mapper {
...
+    ///
+    /// - parameter fields:         The array of fields to retrieve from the source data, can be an empty
+    ///                             string to return the entire data set
+    /// - parameter transformation: The transformation function used to create the expected value
+    ///
+    /// - returns: The value of type T for the given field, if the transformation function doesn't throw
+    ///            otherwise nil
+    public func optionalFrom<T>(_ fields: [String], transformation: (Any) throws -> T?) -> T? {
+        for field in fields {
+            if let value = try? transformation((try self.JSONFromField(field))) {
```
Comment (keith) [inline Sources/Mapper.swift]: remove extra parens?

File: Sources/Mapper.swift [hunk 2]
```diff
@@ -346,6 +346,23 @@ public struct Mapper {
...
+                return value
+            }
+        }
```
Comment (keith) [inline Sources/Mapper.swift]: newline after

**Comment by keith** [comment]
Is there any way we could do this with an implementation for Convertible, instead of just `T`?

**Comment by keith** [comment]
Also please add a changelog entry

File: Tests/MapperTests/CustomTransformationTests.swift [hunk 3]
```diff
@@ -75,4 +75,52 @@ final class CustomTransformationTests: XCTestCase {
...
+            let value: Int?
+            init(map: Mapper) throws {
+                value = map.optionalFrom(["a", "b"], transformation: { _ in
+                    throw MapperError.customError(field: nil, message: "")
+                })
+            }
+        }
+
+        let test = try? Test(map: Mapper(JSON: ["a": "##", "b": 1]))
+        XCTAssertNil(test?.value)
```
Comment (keith) [inline Tests/MapperTests/CustomTransformationTests.swift]: Can you adjust these tests to be like the ones above these new ones, we want to make sure that `test` is non-nil, but value is, so we should catch the `try` with an `XCTFail`

File: Sources/Mapper.swift (hunk 1 above)
Comment (dduan) [reply Sources/Mapper.swift]: done

File: Sources/Mapper.swift (hunk 2 above)
Comment (dduan) [reply Sources/Mapper.swift]: done

File: Tests/MapperTests/CustomTransformationTests.swift (hunk 3 above)
Comment (dduan) [reply Tests/MapperTests/CustomTransformationTests.swift]: done

**Comment by dduan** [comment]
added change log.

**Comment by dduan** [comment]
Now I'll squash.

**Comment by keith** [comment]
Pushed 7.2.0

**Pull request merged**